runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -626,9 +585,13 @@ function STHighlighter:mark_dirty(client_id)
   if state.current_result then
     state.current_result.version = nil
   end
-  local client = vim.lsp.get_client_by_id(client_id)
-  assert(client)
-  self:cancel_all_requests(client)
+
+  if state.active_request.request_id then
```

Comment:
It looks like it might be helpful to abstract this logic of canceling the active client request into a local function.